src/App.vue

```diff
@@ -25,6 +28,15 @@ export default defineComponent({
     Actions,
     Search,
   },
+  setup() {
+    onMounted(() => {
+      if (parse(location.search).payload) {
+        let payload = JSON.parse(parse(location.search).payload);
+        store.dispatch("asana/tokenReceived", payload);
```

Comment:
Does this need an await or is it fine to let it run in the background?

Reply:
It should be fine running in the background.